### PR TITLE
Streaming-independent byte parser

### PR DESCRIPTION
**Pull request by karl-exini**

### CODE REVIEW

File: .github/workflows/release.yml
Comment (karl-exini) [inline .github/workflows/release.yml:1]: Changes to build environment, not really part of this PR

File: .travis.yml
Comment (karl-exini) [inline .travis.yml:1]: Changes to build env. Ignore.

File: src/main/scala/com/exini/dicom/streams/ByteParserFlow.scala [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    override def next(result: T): Unit =
+      push(objOut, result)
+
+    override def fail(ex: Throwable): Unit =
+      failStage(ex)
+
+    override def complete(): Unit =
+      completeStage()
+
+    override def needNoreData(current: ParseStep[T], reader: ByteReader, acceptNoMoreDataAvailable: Boolean): Unit =
```
Comment (pewniak747) [inline src/main/scala/com/exini/dicom/streams/ByteParserFlow.scala]: Typo: `MoreData` ? Shouldn't it be a compile error since it's an `override`?

File: src/test/scala/com/exini/dicom/data/ByteParserTest.scala [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  }
+
+  it should "throw an error for invalid byte sequence" in new Fixture(Seq(ByteString("Car,Boat"))) {
+    assertThrows[DicomParseException] {
+      parse()
+    }
+  }
+
+  it should "accept data in chunks" in new Fixture(
+    Seq(ByteString("STARTBoat,Car"), ByteString("Airplane"))
```
Comment (pewniak747) [inline src/test/scala/com/exini/dicom/data/ByteParserTest.scala:29]: Shouldn't that be `,Airplane` (with comma separating `Car` and `Airplane`)?

**Review by pewniak747 — APPROVED** [review]

File: src/main/scala/com/exini/dicom/streams/ByteParserFlow.scala (hunk 1 above)
Comment (karl-exini) [reply src/main/scala/com/exini/dicom/streams/ByteParserFlow.scala]: Thanks, will fix

File: src/test/scala/com/exini/dicom/data/ByteParserTest.scala (hunk 2 above)
Comment (karl-exini) [reply src/test/scala/com/exini/dicom/data/ByteParserTest.scala:29]: In this particular parser, words can be separated into chunks, or within a chunk by a comma.
